### PR TITLE
docs(button)

### PR DESCRIPTION
**Pull request by gintama91**
### Description

docs trying for button

### Checklist

- [ ] Run tests locally
- [ ] Run linter(check for linter errors)

### CODE REVIEW

**Comment by gintama91** [comment]
ok this is more like , is there anything else we can add here? (for a reference to write other docs)

File: lib/scarpe/button.rb [hunk 1]
```diff
@@ -1,11 +1,46 @@
 # frozen_string_literal: true
 
+# The Scarpe Class
 class Scarpe
+  # The Button class represents a button widget in Scarpe
```
Comment (noahgibbs) [inline lib/scarpe/button.rb:5]: I feel like the top two lines -- "The Scarpe Class" and "The Button class represents a button widget in Scarpe" are probably excessive :-)

File: lib/scarpe/button.rb [hunk 2]
```diff
@@ -1,11 +1,46 @@
...
   class Button < Scarpe::Widget
     include Scarpe::Log
-
+    
     display_properties :text, :width, :height, :top, :left, :size
 
+    # Initializes a new Button object.
```
Comment (noahgibbs) [inline lib/scarpe/button.rb:11]: Nice! This is much better documentation than we've been including. I'll keep working to get it put up where people can see it :-)

**Review by noahgibbs — APPROVED** [review]

**Comment by noahgibbs** [comment]
The test failure isn't your fault there, and looks like one more "hey, it's flaky" problem :-( I'll look into it.

File: lib/scarpe/button.rb (hunk 1 above)
Comment (gintama91) [reply lib/scarpe/button.rb:5]: lol yeah but idk what to write there so i just wrote tht.

Comment (noahgibbs) [reply lib/scarpe/button.rb:5]: For this, it's probably okay to just not write anything until there's something to write. Somebody seeing "Button < Scarpe::Widget" is gonna figure out that it's a class representing the button as a Scarpe widget.

**Comment by gintama91** [comment]
making  pr for every widget docs is probably excessive so i will just do in this one , taking this one as reference for other widgets too

**Comment by noahgibbs** [comment]
Sounds good. Feel free to add as many in a single PR as you like.

**Comment by gintama91** [comment]
ah ig this got closed when i was trying to update branch

File: lib/scarpe/para.rb [hunk 3]
```diff
@@ -10,19 +10,74 @@ def inherited(subclass)
...
+    #
+    #       banner_widget = banner("Welcome to Shoes!")
+    #       title_widget = title("Shoes Examples")
+    #       subtitle_widget = subtitle("Explore the Features")
+    #       tagline_widget = tagline("Step into a World of Shoes")
+    #       caption_widget = caption("A GUI Framework for Ruby")
+    #       inscription_widget = inscription("Designed for Easy Development")
+    #
+    #      end
+    #
```
Comment (gintama91) [inline lib/scarpe/para.rb:64]: hm i am thinking to add image but i cant really figure out relative path and all or what to use i tried using , didnt worked

Comment (noahgibbs) [reply lib/scarpe/para.rb:64]: Yeah, may be hard to get a runnable docs example with a local image. Pretty much depends on them having the right image in the right place.

This is also a little verbose -- keep in mind that these are going to be API reference docs, not a full manual. You can *have* manual pages, but they shouldn't be embedded in the widget code.

Comment (noahgibbs) [reply lib/scarpe/para.rb:64]: What you want here is more like the manual entry for para, not like how "Nobody Knows Shoes" describes para.

**Pull request closed without merging**